models/sessions.go

```diff
@@ -129,8 +129,16 @@ func FindSessionByUserID(tx *storage.Connection, userId uuid.UUID) (*Session, er
 	return session, nil
 }
 
+func FindSessionsByFactorID(tx *storage.Connection, factorID uuid.UUID) ([]*Session, error) {
+	sessions := []*Session{}
+	if err := tx.Q().Where("factor_id = ?", factorID).All(&sessions); err != nil {
+		return nil, err
+	}
+	return sessions, nil
+}
+
 func updateFactorAssociatedSessions(tx *storage.Connection, userID, factorID uuid.UUID, aal string) error {
-	return tx.RawQuery("UPDATE "+(&pop.Model{Value: Session{}}).TableName()+" set aal = ?, factor_id = ? WHERE user_id = ? AND factor_id = ?", aal, uuid.Nil, userID, factorID).Exec()
+	return tx.RawQuery("UPDATE "+(&pop.Model{Value: Session{}}).TableName()+" set aal = ?, factor_id = ? WHERE user_id = ? AND factor_id = ?", aal, nil, userID, factorID).Exec()
```

Comment:
use `nil` over `uuid.Nil` so the `factor_id` will be set to `null` instead of a null uuid - this is consistent with the initial creation of a session where the `factor_id` is null